modules/api_reference/src/Exception/DpApiReferenceException.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Drupal\devportal_api_reference\Exception;
+
+/**
+ * Defines exception for Devportal API Reference.
+ */
+class DpApiReferenceException extends \Exception { }
```

Comment:
These should be in seperate files ofc. 

```php
<?php

namespace Drupal\devportal_api_reference\Exception;

/**
 * Module specific base exception.
 */
interface ApiReferenceException extends \Throwable {

}

/**
 * Module specific Runtime exception.
 */
class RuntimeException extends \RuntimeException implements ApiReferenceException {

}

/**
 * Module specific InvalidArgument exception.
 */
class InvalidArgumentException extends \InvalidArgumentException implements ApiReferenceException {

}

/**
 * Thrown if the input file could not be parsed.
 */
class ParseException extends RuntimeException {
    
    /** @var string */
    private $fileType;

    public function __construct(string $file_type, string $path, string $reason, int $code = null, \Throwable $previous_exception = null)
    {
        $file_type = strtoupper($file_type);
        $this->fileType = $file_type;
        $code = $code ?? 0;
        parent::__construct("The {$file_type} source file at {$path} path could not be parsed because: {$reason}.", $code, $previous_exception);
    }

    public static function yamlParseError(string $path, string $reason, \Throwable $previous_exception = null) {
        return new static('YAML', $path, $reason, null, $previous_exception);
    }

    public static function jsonParseError(string $path, int $json_last_error_code, string $json_last_error_message, \Throwable $previous_exception = null) {
        return new static('JSON', $path, $json_last_error_message, $json_last_error_code, $previous_exception);
    }

    /**
     * @return string
     */
    public function getFileType(): string
    {
        return $this->fileType;
    }
}


```